test/setup/build.ts

```diff
@@ -0,0 +1,35 @@
+import { execSync } from 'child_process';
+import { renameSync, rmSync } from 'fs';
+import { projectRoot } from '../utils';
+
+const rustGithubRepository = 'metaplex-program-library';
+const repostitoryDir = `${rustGithubRepository}-master`;
+
+const rustProgramsRepository = `https://github.com/metaplex-foundation/${rustGithubRepository}/archive/refs/heads/master.zip`;
+
+function build() {
+  const programs: string[] = [
+    'auction/program',
+    'token-metadata/program',
+    'token-vault/program',
+    'metaplex/program',
+  ];
+
+  process.chdir('test');
+  rmSync(repostitoryDir, { recursive: true, force: true });
+  rmSync('rust', { recursive: true, force: true });
+  rmSync('./master.zip', { force: true });
+
+  execSync(`wget --quiet ${rustProgramsRepository}`);
+  execSync('unzip master.zip');
+
+  programs.forEach((directory) => {
+    process.chdir(`${projectRoot}/test/${repostitoryDir}/${directory}`);
```

Comment:
ditto regarding the `chdir`. I'm pretty sure there must be a way to provide the project directory to cargo. If that option does not exist then I suppose `chdir` is fine here, but then you should store the current directory and change back to it after you're done. Otherwise things get confusing very quickly.